test/parallel/test-require-package-errors.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+require('../common');
+
+const assert = require('assert');
+const path = require('path');
+const fs = require('fs');
+const tmpdir = require('../common/tmpdir');
+
+tmpdir.refresh();
+
+fs.writeFileSync(path.join(tmpdir.path, 'index.js'), 'require("a");');
+fs.mkdirSync(path.join(tmpdir.path, 'node_modules'));
+fs.mkdirSync(path.join(tmpdir.path, 'node_modules/a'));
+fs.mkdirSync(path.join(tmpdir.path, 'node_modules/b'));
+fs.writeFileSync(path.join(tmpdir.path, 'node_modules/a/a.js'), 'require("b");');
+fs.writeFileSync(path.join(tmpdir.path, 'node_modules/a/package.json'), '{ "main": "a.js" }');
+fs.writeFileSync(path.join(tmpdir.path, 'node_modules/b/package.json'), '{ "main": "b.js" }');
+
+let err;
+try {
+  require(path.join(tmpdir.path, 'index.js'));
+} catch (e) {
+  err = e;
+}
+
+assert.deepStrictEqual(err.requireStack, [ '/path/to/tmpdir/node_modules/a/a.js', '/path/to/tmpdir/index.js']);
```

Comment:
The test fails due to not yet having the `requireStack` property.

Besides that, please also use `assert.throws()` instead of the try catch:

```suggestion
assert.throws(() => {
  require(path.join(tmpdir.path, 'index.js'));
}, {
  requireStack: ['/path/to/tmpdir/node_modules/a/a.js', '/path/to/tmpdir/index.js'],
});
```